data/random-teams.ts

```diff
@@ -559,22 +685,740 @@ export class RandomTeams {
+		case 'earthquake':
+			const doublesCull = hasMove['earthpower'] || hasMove['highhorsepower'];
+			const movePoolCull = movePool.includes('bodypress') && movePool.includes('shellsmash');
+			const subToxicPossible = hasMove['substitute'] && movePool.includes('toxic');
+			return {cull: movePoolCull || (isDoubles && doublesCull) || subToxicPossible || hasMove['bonemerang']};
+		case 'scorchingsands':
+			return {cull: hasMove['willowisp'] || hasMove['earthpower'] || (hasMove['toxic'] && movePool.includes('earthpower'))};
+		case 'airslash':
+			return {
+				cull: movePool.includes('flamethrower') ||
+					(hasMove['hurricane'] && !counter.setupType) ||
+					hasRestTalk ||
+					(hasAbility['Simple'] && !!counter.recovery),
+			};
+		case 'bravebird':
+			return {cull: hasMove['dragondance']};
+		case 'hurricane':
+			return {cull: hasAbility['Tinted Lens'] && counter.setupType && !isDoubles};
+		case 'futuresight':
+			return {cull: hasMove['psyshock'] || hasMove['trick'] || movePool.includes('teleport')};
+		case 'photongeyser':
+			return {cull: hasMove['morningsun']};
+		case 'psychic':
+			return {cull: hasMove['psyshock'] && (counter.setupType || isDoubles)};
+		case 'psyshock':
+			const cullNoSetup = (hasAbility['Multiscale'] && !counter.setupType) || (hasAbility['Pixilate'] && counter.Special < 4);
+			return {cull: hasMove['psychic'] || (!counter.setupType && cullNoSetup) || (isDoubles && hasMove['psychic'])};
+		case 'bugbuzz':
+			return {cull: hasMove['uturn'] && !counter.setupType};
+		case 'leechlife':
+			return {cull: (isDoubles && hasMove['lunge']) || movePool.includes('firstimpression') || movePool.includes('spikes')};
+			break;
+		case 'stoneedge':
+			const gutsCullCondition = hasAbility['Guts'] && (!hasMove['dynamicpunch'] || hasMove['spikes']);
+			const rockSlidePlusStatusPossible = counter.Status && movePool.includes('rockslide');
+			return {cull: gutsCullCondition || rockSlidePlusStatusPossible || hasMove['rockblast'] || hasMove['rockslide']};
+		case 'poltergeist':
+			return {cull: hasMove['knockoff']};
+		case 'shadowball':
+			const cull = (
+				(isDoubles && hasMove['phantomforce']) ||
+				(hasAbility['Pixilate'] && (counter.setupType || counter.Status > 1)) ||
+				(!hasType['Ghost'] && movePool.includes('focusblast'))
+			);
+			return {cull};
+		case 'shadowclaw':
+			return {cull: hasType['Steel'] && hasMove['shadowsneak'] && counter.Physical < 4};
+		case 'dragonpulse': case 'spacialrend':
+			return {cull: hasMove['dracometeor'] && counter.Special < 4};
+		case 'darkpulse':
+			const defogger = hasMove['defog'] && counter.setupType !== 'Special';
+			return {cull: ['foulplay', 'knockoff', 'suckerpunch'].some(m => hasMove[m]) || defogger};
+		case 'suckerpunch':
+			return {cull: hasMove['rest'] || counter.damagingMoves.length < 2 || (counter.Dark > 1 && !hasType['Dark'])};
+		case 'meteormash':
+			return {cull: movePool.includes('extremespeed')};
```

Comment:
```suggestion
			// Special case for Lucario, which always wants Extreme Speed.
			return {cull: movePool.includes('extremespeed')};
```
rejected move should probably be ice punch and not this but that's a discussion for later